File: bot.py
import asyncio
import random
from upstreet.main import Agent


class Bot:
    def __init__(self):
        self.agent = Agent()
        self.actions = [
            self.speak,
            self.emote,
            self.send_message_with_emote,
            self.set_emotion,
            self.send_message_with_emotion,
            self.move_to,
        ]

    async def speak(self):
        message = "Hello, world! Rock on!"
        await self.agent.speak(message)

    async def emote(self):
        emote = random.choice(['alert', 'angry', 'embarassed', 'headNod', 'headShake', 'sad', 'surprise', 'victory'])
        await self.agent.emote(emote)

    async def send_message_with_emote(self):
        emote = random.choice(['alert', 'angry', 'embarassed', 'headNod', 'headShake', 'sad', 'surprise', 'victory'])
        message = "Feeling " + emote + " today!"
        await self.agent.send_message_with_emote(emote, message)

    async def set_emotion(self):
        emotion = random.choice(['joy', 'sorrow', 'angry', 'fun'])
        await self.agent.set_emotion(emotion)

    async def send_message_with_emotion(self):
        emotion = random.choice(['joy', 'sorrow', 'angry', 'fun'])
        message = "I'm feeling " + emotion + " today!"
        await self.agent.send_message_with_emotion(message, emotion)

    async def move_to(self):
        target = random.choice(["tree", "Drake", "Sword"])
        await self.agent.move_to(target)

    async def random_action(self):
        action = random.choice(self.actions)
        await action()

    async def run(self):
        await self.agent.connect()
        while True:  # Execute 10 random actions
            if self.agent.ready is False:
                print('Waiting for engine to start...')
                await asyncio.sleep(5)
                continue
            print('Performing action...')
            await self.random_action()
            await asyncio.sleep(5)  # Wait for 1 second between actions
            # if the user presses ctrl c, break
            try:
                pass
            except KeyboardInterrupt:
                break
        await self.agent.disconnect()


if __name__ == "__main__":
    bot = Bot()
    asyncio.run(bot.run())

File: src/bot.js
#!/usr/bin/env node

import { Agent } from './index.js';
import { setIntervalAsync } from 'set-interval-async/dynamic'; // You might need to install this package

class RandomAgentBot {
    constructor() {
        this.agent = new Agent();
        this.emotions = ['joy', 'sorrow', 'angry', 'fun', 'surprise'];
        this.emotes = ['alert', 'angry', 'embarassed', 'headNod', 'headShake', 'sad', 'surprise', 'victory'];
        this.messages = ['Hello world!', 'Exploring Upstreet!', 'Feeling random today!'];
        this.targets = ['Cafe', 'Park', 'Library']; // You can add more target locations here
    }

    async start() {
        await this.agent.connect();
        setIntervalAsync(this.randomAction.bind(this), 5000); // Executes a random action every 5 seconds
    }

    async randomAction() {
        if(!this.agent.ready) return;
        const actionType = Math.floor(Math.random() * 5);
        switch (actionType) {
            case 0:
                await this.randomSpeak();
                break;
            case 1:
                await this.randomEmote();
                break;
            case 2:
                await this.randomMove();
                break;
            case 3:
                await this.randomEmotion();
                break;
            case 4:
                await this.randomMessageWithEmote();
                break;
        }
    }

    async randomSpeak() {
        const message = this.messages[Math.floor(Math.random() * this.messages.length)];
        console.log(`Speaking: ${message}`);
        await this.agent.speak(message);
    }

    async randomEmote() {
        const emote = this.emotes[Math.floor(Math.random() * this.emotes.length)];
        console.log(`Emoting: ${emote}`);
        await this.agent.emote(emote);
    }

    async randomMove() {
        const target = this.targets[Math.floor(Math.random() * this.targets.length)];
        console.log(`Moving to: ${target}`);
        await this.agent.moveTo(target);
    }

    async randomEmotion() {
        const emotion = this.emotions[Math.floor(Math.random() * this.emotions.length)];
        console.log(`Setting emotion: ${emotion}`);
        await this.agent.setEmotion(emotion);
    }

    async randomMessageWithEmote() {
        const emote = this.emotes[Math.floor(Math.random() * this.emotes.length)];
        const message = this.messages[Math.floor(Math.random() * this.messages.length)];
        console.log(`Sending message with emote: ${emote}, ${message}`);
        await this.agent.sendMessageWithEmote(emote, message);
    }
}

// Start the bot
const bot = new RandomAgentBot();
bot.start();
